release.py

```diff
@@ -0,0 +1,243 @@
+#!/usr/bin/env python3
+
+from __future__ import annotations
+
+"""
+Tool to help automate changes needed in commits during and after releases
+"""
+
+import argparse
+import logging
+import sys
+from datetime import datetime
+from pathlib import Path
+from subprocess import PIPE, run
+from typing import List
+
+LOG = logging.getLogger(__name__)
+NEW_VERSION_CHANGELOG_TEMPLATE = """\
+## Unreleased
+
+### Highlights
+
+<!-- Include any especially major or disruptive changes here -->
+
+### Stable style
+
+<!-- Changes that affect Black's stable style -->
+
+### Preview style
+
+<!-- Changes that affect Black's preview style -->
+
+### Configuration
+
+<!-- Changes to how Black can be configured -->
+
+### Packaging
+
+<!-- Changes to how Black is packaged, such as dependency requirements -->
+
+### Parser
+
+<!-- Changes to the parser or to version autodetection -->
+
+### Performance
+
+<!-- Changes that improve Black's performance. -->
+
+### Output
+
+<!-- Changes to Black's terminal output and error messages -->
+
+### _Blackd_
+
+<!-- Changes to blackd -->
+
+### Integrations
+
+<!-- For example, Docker, GitHub Actions, pre-commit, editors -->
+
+### Documentation
+
+<!-- Major changes to documentation and policies. Small docs changes
+     don't need a changelog entry. -->
+"""
+
+
+class NoGitTagsError(Exception): ...  # noqa: E701,E761
+
+
+# TODO: Do better with alpha + beta releases
+# Maybe we vendor packaging library
+def get_git_tags(versions_only: bool = True) -> List[str]:
+    """Pull out all tags or calvers only"""
+    cp = run(["git", "tag"], stdout=PIPE, stderr=PIPE, check=True, encoding="utf8")
+    if not cp.stdout:
+        LOG.error(f"Returned no git tags stdout: {cp.stderr}")
+        raise NoGitTagsError
+    git_tags = cp.stdout.splitlines()
+    if versions_only:
+        return [t for t in git_tags if t[0].isdigit()]
+    return git_tags
+
+
+def int_calver(calver: str) -> int:
+    """Convert a calver string into an hex base 16 integer for sorting
```

Comment:
I don't think this does hex, not sure what the docstring is talking about.

Also, won't this sort incorrectly in some cases? 23.10.0 -> 23100, 24.1.0 -> 2410, the second is smaller.

I think we can just do something like `tuple(map(int, calver.split(".")))`.